app/assets/javascripts/views/patient_builder/data_criteria.js.coffee

```
@@ -195,12 +195,19 @@ class Thorax.Views.EditCriteriaView extends Thorax.Views.BuilderChildView
 class Thorax.Views.CodeSelectionView extends Thorax.Views.BuilderChildView
   template: JST['patient_builder/edit_codes']
   events:
-    'change select':           'validateForAddition'
-    'change .codeset-control': 'changeConcepts'
+    'change select[name=codeset]': (e) ->
+      @model.set codeset: $(e.target).val()
+      @changeConcepts(e)
+      @validateForAddition()
+    'change select[name=code]' : ->
+      @validateForAddition()
+    'keyup input[name=custom_code]': 'validateForAddition'
+    'keyup input[name=custom_codeset]': 'validateForAddition'
     rendered: ->
       @$('select.codeset-control').selectBoxIt('native': true)
 
   initialize: ->
+    @custom_key = 'custom'
```

I'm not sure I see the value in having a variable to hold the string 'custom', rather than just using the string 'custom' in the appropriate places. If the goal is to prevent silent failures if one of the instances is changed without changing the others, the use of the variable still has that issue because of how handlebars templates work (they'll silently handle whatever is there even if it's not defined, i think).